src/main.rs

```diff
@@ -29,41 +29,56 @@ fn force_update_async() -> Result<()> {
 }
 
 // Runs bkt after main() handles flag parsing
-#[allow(clippy::too_many_arguments)]
-fn run(root_dir: Option<PathBuf>, discard_failures: bool, scope: Option<&str>,
-       mut command: CommandDesc, use_cwd: bool, env_keys: BTreeSet<&OsStr>, ttl: Duration,
-       stale: Option<Duration>, warm: bool, force: bool) -> Result<i32> {
-    assert!(!ttl.as_secs() > 0 || ttl.subsec_nanos() > 0, "--ttl cannot be zero"); // TODO use is_zero once stable
+fn run(cli: Cli) -> Result<i32> {
+    let ttl: Duration = cli.ttl.into();
+    let stale: Option<Duration> = cli.stale.map(Into::into);
+
+    assert!(
+        !ttl.as_secs() > 0 || ttl.subsec_nanos() > 0,
+        "--ttl cannot be zero"
+    ); // TODO use is_zero once stable
     if let Some(stale) = stale {
-        assert!(!stale.as_secs() > 0 || stale.subsec_nanos() > 0, "--stale cannot be zero"); // TODO use is_zero once stable
+        assert!(
+            !stale.as_secs() > 0 || stale.subsec_nanos() > 0,
+            "--stale cannot be zero"
+        ); // TODO use is_zero once stable
         assert!(stale < ttl, "--stale must be less than --ttl");
     }
 
-    let mut bkt = match root_dir {
+    let mut bkt = match cli.cache_dir {
         Some(cache_dir) => Bkt::create(cache_dir)?,
         None => Bkt::in_tmp()?,
     };
-    bkt = bkt.discard_failures(discard_failures);
-    if let Some(scope) = scope {
-        bkt = bkt.scoped(scope.into());
+    bkt = bkt.discard_failures(cli.discard_failures);
+    if let Some(scope) = cli.scope {
+        bkt = bkt.scoped(scope);
     }
 
-    if use_cwd {
+    let mut command = CommandDesc::new(cli.command);
+
+    if cli.cwd {
         command = command.with_cwd()?;
     }
+
+    let env_keys: BTreeSet<_> = cli
```

Comment:
Similarly I'd just put this on one line. I know it may be more typical to flow into multiple lines, but until/unless I configure a formatter for the repo I'm not going to bother.